lib/DDG/Meta/Data.pm

```diff
@@ -36,91 +34,74 @@ unless(%ia_metadata){
         mkdir $mdir or die "Failed to mkdir $mdir: $!";
     }
 
-    # Load IA metadata. Not all types are required during development.
-    for my $iat (@ia_types){
-        debug && warn "Processing IA type: $iat";
-
-        my $json_endpt = lc $iat;
-        $json_endpt =~ s/goodie/goodies/;
+    debug && warn "Processing metadata";
 
-        # Prefer freshly downloaded metadata and fall back to metadata
-        # bundled with installed IA repos
-        my $f = "$mdir/$iat.json";
-        my $c = $ENV{NO_METADATA_DOWNLOAD} ? 0 : mirror("https://duck.co/ia/repo/$json_endpt/json", $f);
-        my $json;
-        if($c == RC_OK || $c == RC_NOT_MODIFIED){
-            $json = io($f)->slurp; 
+    my $f = "$mdir/metadata.json";
+    unless($ENV{NO_METADATA_DOWNLOAD}){
+        my $ua = LWP::UserAgent->new;
+        $ua->timeout(5);
+        $ua->default_header('Accept-Encoding' => scalar HTTP::Message::decodable());
+        my $res = $ua->mirror("https://duck.co/ia/repo/all/json", $f);
+        unless($res->is_success){
+            debug && warn "Failed to download metdata: " . $res->status_line;
         }
-        else {
-            debug && warn "Failed to download metdata for $iat: $c";
-            if(-f $f){
-                $json = io($f)->all;
-            }
-            else{
-               warn "Failed to download metadata for $iat and no local file $f";
-            }
-        }
-
-        next unless $json;
-        my $metadata = eval{ decode_json($json); } or warn "Failed to decode_json: $@";
-        next unless $metadata;
+    }
 
-        # One metadata file for each repo with the following format
-        # { "<IA name>": {
-        #     "id": " " 
-        #     "signal" : " "
-        #     ....
-        # }
+    my $metadata = decode_json(io($f)->all);
```

Comment:
We want it to die here.  Before we were downloading four different files so only one of them was "necessary".